src/helm/benchmark/metrics/per_request_metrics.py

```diff
@@ -0,0 +1,260 @@
+from typing import Dict, List, Optional
+
+import json
+import importlib_resources as resources
+
+from helm.common.hierarchical_logger import hlog
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.benchmark.adaptation.adapter_spec import AdapterSpec
+from helm.benchmark.window_services.window_service import WindowService
+from helm.benchmark.window_services.window_service_factory import WindowServiceFactory
+from helm.benchmark.window_services.tokenizer_service import TokenizerService
+from .metric_name import MetricName
+from .metric_service import MetricService
+from .statistic import Stat
+
+
+EFFICIENCY_DATA_PACKAGE: str = "helm.benchmark.efficiency_data"
+
+INFERENCE_IDEALIZED_RUNTIMES_JSON_FILENAME: str = "inference_idealized_runtimes.json"
+INFERENCE_DENOISED_RUNTIMES_JSON_FILENAME: str = "inference_denoised_runtimes.json"
+TRAINING_EFFICIENCY_JSON_FILENAME: str = "training_efficiency.json"
+
+
+# TODO Actually make this work like a Metric. The current state is just trying to split
+# it out of other Metrics to make refactoring easier.
+class PerRequestMetric:
+    def __init__(self):
+        # For Efficiency metrics:
```

Comment:
I think it would make sense to put all the efficiency metrics in a separate file `efficiency_metrics.py`.  I don't think this code is actively being used (e.g., with HELM lite).

Reply:
Done. I moved the non efficiency logic back into basic_metrics.py, but kept them out of BasicMetric. I have a forthcoming refactor related to language modeling which needs them to be separate.